tests/functional/adapter/materialization/test_table_drop.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal
+
+from tests.functional.adapter.materialization.fixtures import (
+    seed_csv,
+    model_sql,
+    profile_yml,
+)
+
+
+class TestTableDrop:
+    """
+    Testing on_table_exists = `drop` configuration for table materialization,
+    using dbt seed, run and tests commands and validate data load correctness.
+    """
+
+    # configuration in dbt_project.yml
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "example",
+            "models": {
+                "+materialized": "table",
+                "+on_table_exists": "drop"
+            }
+        }
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "seed.csv": seed_csv,
+        }
+
+    # everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "materialization.sql": model_sql,
+            "materialization.yml": profile_yml,
+        }
+
+    # The actual sequence of dbt commands and assertions
+    # pytest will take care of all "setup" + "teardown"
+    def test_run_seed_test(self, project):
+        # seed seeds
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+        # run models two times to check on_table_exists = 'drop'
```

Comment:
https://github.com/dbt-labs/dbt-core/blob/c270a77552ae9fc66fdfab359d65a8db1307c3f3/tests/adapter/dbt/tests/adapter/basic/test_incremental.py

We could use the same mechanism as in test-incremental  to change the seed before running the second time. This way, the test looks more realistic.